packages/react-hooks/src/RecordProvider.tsx

```diff
@@ -0,0 +1,108 @@
+import type { Agent, RecordDeletedEvent, RecordSavedEvent, RecordUpdatedEvent, BaseRecord } from '@aries-framework/core'
+import type { Constructor } from '@aries-framework/core/build/utils/mixins'
+
+import { RepositoryEventTypes } from '@aries-framework/core'
+import * as React from 'react'
+import { createContext, useContext, useEffect, useState } from 'react'
+
+export interface RecordContextInterface {
+  loading: boolean
+  records: BaseRecord[]
+}
+
+interface Props {
+  agent: Agent | undefined
+}
+
+const RecordContext = createContext<RecordContextInterface | undefined>(undefined)
+
+export const useRecords = () => {
+  const recordContext = useContext(RecordContext)
+  if (!recordContext) {
+    throw new Error('useRecords must be used within a RecordContextProvider')
+  }
+  return recordContext
+}
+
+export const useRecordsByType = <R extends BaseRecord<any, any, any>>(
+  recordClass: Constructor<R> & { type: string }
+): R[] => {
+  const { records } = useRecords()
+  return records.filter((record) => record.type === recordClass.type) as R[]
+}
+
+const RecordProvider: React.FC<Props> = ({ agent, children }) => {
+  const [recordState, setRecordState] = useState<RecordContextInterface>({
+    loading: true,
+    records: [],
+  })
+
+  const setInitialState = async () => {
+    if (agent) {
+      const records = [
+        ...((await agent.connections.getAll()) as unknown as BaseRecord[]),
```

Comment:
@TimoGlastra I don't suppose there is a module for getting base records?

Reply:
No we need the class and record type to correctly retrieve records from storage...

Reply:
Ok. So does this approach work then? It really obviates the need for the individual providers. I have just extended the existing hooks.

Reply:
That was my concern, that we can no longer subscribe to individual events or record types. Every hook now filters through every record and receives an update for any record type (I suppose this is not what we want)